File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <bits/stdc++.h>
using namespace std;

set<int> s;


int solution(vector<int> elements) {
    vector<int> v = elements;
    for(int i = 0; i<elements.size(); i++){
        v.push_back(elements[i]);
    }

    int i = 1;


    while(i < elements.size()){
        for(int j = 0; j<elements.size(); j++){
            int sum = 0;
            for(int k = j; k<j+i; k++){
                sum += v[k];
            }
            s.insert(sum);
        }
        i++;
    }
    
    return s.size()+1;
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.HashSet;
import java.util.Set;

public class Solution {

    public int solution(int[] elements) {
        int n = elements.length;
        int[] doubled = new int[2 * n];
        for (int i = 0; i < n; i++) {
            doubled[i] = elements[i];
            doubled[i + n] = elements[i];
        }

        Set<Integer> sumSet = new HashSet<>();
        for (int len = 1; len <= n; len++) {
            int sum = 0;
            for (int i = 0; i < len; i++) {
                sum += doubled[i];
            }
            sumSet.add(sum);
            for (int i = 1; i < n; i++) {
                sum = sum - doubled[i - 1] + doubled[i + len - 1];
                sumSet.add(sum);
            }
        }
        return sumSet.size();
    }

}
